File: arena/spec/problem_objects/array_read_swap_spec.lua
local mach = require'mach'
local problem_object = require'problem_objects.array_read_swap'

describe('array_read_swap', function()

  it('should create an array that can be read from and swapped from only', function()
    local arr = problem_object({'hello', 'goodbye', 'iloveyou'})

    assert.are_equal('hello', arr.read(1))
    arr.swap(1, 3)
    assert.are_equal('iloveyou', arr.read(1))
  end)

  it('should error if an index is swapped that does not have data', function()
    local arr = problem_object({'hello'})

    assert.has_error(function() arr.swap(1,2) end, 'tried to swap from an index that has no data')
  end)
end)

File: arena/problem_object/spec/array_read_swap_spec.dart
import "package:test/test.dart";
import "../src/array_read_swap.dart";

void main() {
  test("should be able to read at an index", () {
    var problem_object = new array_read_swap([1,2,3,4]);

    expect(problem_object.read(0), equals(1));
  });

  test("should be able to swap two indexes", () {
    var problem_object = new array_read_swap([1,2,3,4]);

    problem_object.swap(2,3);
    expect([1,2,4,3], equals(problem_object.array));
  });
}
